### PR TITLE
Adding enhancements for GraphQL

### PR DESCRIPTION
**Pull request by kikoluiss**
- Type GraphQL: api_management_api
- Definitions and Components: api_management_api_schema

### CODE REVIEW

File: website/docs/r/api_management_api.html.markdown [hunk 1]
```diff
@@ -82,6 +82,8 @@ The following arguments are supported:
 
 * `subscription_required` - (Optional) Should this API require a subscription key?
 
+* `type` - (Optional) The type of the API: "http", "graphql", "soap", "websocket"
```
Comment (katbyte) [inline website/docs/r/api_management_api.html.markdown:85]: should this be `api_type` to make sure its exceptionally clear what "type" we are referring to?

File: website/docs/d/api_management_api.html.markdown [hunk 2]
```diff
@@ -59,6 +59,8 @@ output "api_management_api_id" {
 
 * `subscription_required` - Should this API require a subscription key?
 
+* `type` - The type of the API: "http", "graphql", "soap", "websocket"
```
Comment (katbyte) [inline website/docs/d/api_management_api.html.markdown:62]: should this be `api_type` to make sure its exceptionally clear what "type" we are referring to?

File: internal/services/apimanagement/api_management_api_data_source.go [hunk 3]
```diff
@@ -79,6 +79,12 @@ func dataSourceApiManagementApi() *pluginsdk.Resource {
 				Computed: true,
 			},
 
+			"type": {
```
Comment (katbyte) [inline internal/services/apimanagement/api_management_api_data_source.go:82]: missed renaming this one

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by katbyte** [comment]
thanks @kikoluiss - have some test failures now
image

**Comment by kikoluiss** [comment]
I will review this issue tonight! Regards.

**Review by katbyte — CHANGES_REQUESTED** [review]
still seeing some failures related to api_type
image

**Pull request closed without merging**